Source/EasyNetQ/Consumer/InternalConsumer.cs

```diff
@@ -137,6 +137,7 @@ public void HandleModelShutdown(object model, ShutdownEventArgs reason)
                 Queue.Name,
                 reason
             );
+            Cancel();
```

Comment:
I propose to move `Cancel();` before `logger.InfoFormat()` to inline behavior with `HandleBasicCancel()`.

Reply:
Also just FYI regarding `Cancel()`:

```c#
 // copy to temp variable to be thread safe.
 var localCancelled = cancelled;
```
Copying is superfluous, thread safety does not an issue here because `cancelled` parameter is passed through the constructor and does not change later, only `ConsumerCancelled` event can change.

Reply:
This copy seems to be necessary only for the null checking purpose but `?.Invoke` is already doing it, so we could remove all copies completely.

Reply:
Agree.